src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/StorePal.cs

```diff
@@ -21,19 +21,15 @@ namespace Internal.Cryptography.Pal
 {
     internal sealed partial class StorePal : IDisposable, IStorePal
     {
-        public IEnumerable<X509Certificate2> Certificates
+        public void CopyTo(X509Certificate2Collection collection)
         {
-            get
-            {
-                LowLevelListWithIList<X509Certificate2> certificates = new LowLevelListWithIList<X509Certificate2>();
+            Debug.Assert(collection != null);
 
-                SafeCertContextHandle pCertContext = null;
-                while (Interop.crypt32.CertEnumCertificatesInStore(_certStore, ref pCertContext))
-                {
-                    X509Certificate2 cert = new X509Certificate2(pCertContext.DangerousGetHandle());
-                    certificates.Add(cert);
-                }
-                return certificates;
+            SafeCertContextHandle pCertContext = null;
+            while (Interop.crypt32.CertEnumCertificatesInStore(_certStore, ref pCertContext))
+            {
+                X509Certificate2 cert = new X509Certificate2(pCertContext.DangerousGetHandle());
+                collection.Add(cert);
```

Comment:
The only danger I see here is if it's possible for an exception to get thrown such that we would have added only some of the certs to the output collection and then if there's code that tries to recover. I don't know if that's a real concern.